### DISCUSSION TITLE
Choosing the specific instance of texts (pixi)

### DISCUSSION THREAD

**Discussion by nalcakar** [pixijs/pixijs#8909] [Q&A]
**Hi, I am new to pixi and javascript? I have a question?
When I click any of texts I created through for, only the last instance goes to target x. But I want to move the instance I click on. How can I do this?**

<!doctype html>
<html>
  <head>
    <script src="https://pixijs.download/release/pixi.js"></script>
  </head>
  <body>
    <p id="demo"></p>
    <script>
const app = new PIXI.Application({ background: '#1099bb' });
    document.body.appendChild(app.view);

    for (let i = 0; i < 7 ; i++) {
  
  var bunny = new PIXI.Text("ilker");
    
    bunny.x =  10;
    bunny.y = 50*i;
    bunny.interactive = true;
    bunny.cursor = 'pointer';
      
   
    app.stage.addChild(bunny);

    bunny. On('pointerdown', onClick);
       
function onClick() {
    bunny.x = 50;
    bunny.y = 400;
    
 }
  
}

    </script>
    

  </body>
  </html>

**Comment by SuperSodaSea** [comment]
Use `let bunny` instead of `var bunny`.

[`var`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var) is **function-scoped** or **globally-scoped**, so your code is actually treated like this:

```js
var bunny;
for (let i = 0; i < 7 ; i++) {
    bunny = new PIXI.Text(...);
    // ...
```

There is only one `bunny` variable over all iteration of the loop. After the loop is finished, the `bunny` will be the last `Text`, causing the problem you see.

[`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) is **block-scoped**, for every iteration in the loop, there will be a different `bunny` variable, so it will work as intended.

In almost all situations, it will be a good idea to use `let` over `var` to prevent potential problems.

See also: [What is the difference between "let" and "var"?](https://stackoverflow.com/questions/762011/what-is-the-difference-between-let-and-var)

**Reply by domis86** [reply]
what about :

```
function onClick() {
bunny.x = 50;
bunny.y = 400;

}
```
Doesnt this create other problems? 

Shouldnt `this` be used in such function instead of some `bunny` object? See function `onButtonDown` in https://pixijs.io/examples/#/events/interactivity.js

**Reply by SuperSodaSea** [reply]
Yeah, using `this.x` and `this.y` is also a good way to solve this problem.

**Comment by nalcakar** [comment]
thanks for all your suggestions. it worked.
